http/logical.go

```diff
@@ -364,9 +365,23 @@ func handleLogicalInternal(core *vault.Core, injectDataIntoTopLevel bool, noForw
 				nsPath = ""
 			}
 			if strings.HasPrefix(r.URL.Path, fmt.Sprintf("/v1/%ssys/events/subscribe/", nsPath)) {
-				handler := handleEventsSubscribe(core, req)
-				handler.ServeHTTP(w, r)
-				return
+				// Only the leader handles event subscriptions.
+				// It is safe to call core.Leader() since we have not yet grabbed the stateLock.
+				isLeader, _, _, err := core.Leader()
```

Comment:
I tried testing this out locally, and it didn't seem to work - there's another earlier request forwarding handler that seems to intercept the request before it gets here for non-leaders. Have you been able to get this working locally? Not sure if my test case is flawed (a9d1517065ab8ac92d73fd743f5de6c3bebc9386), but either way it would be good to get an automated test into this PR.

Reply:
Good call. I'll work on getting a good test for this. Thanks for the catch.